types/block.go

```diff
@@ -226,121 +227,53 @@ func (b *Block) fillHeader() {
 func (b *Block) fillDataAvailabilityHeader() {
 	namespacedShares, dataSharesLen := b.Data.ComputeShares()
 	shares := namespacedShares.RawShares()
+
 	if len(shares) == 0 {
 		// no shares -> no row/colum roots -> hash(empty)
 		b.DataHash = b.DataAvailabilityHeader.Hash()
 		return
 	}
 
+	// create the nmt wrapper to generate row and col commitments
+	squareSize := uint32(math.Sqrt(float64(len(shares))))
+	tree := wrapper.NewErasuredNamespacedMerkleTree(uint64(squareSize))
+
```

Comment:
this is the a major reason why we had to move the wrapper and constants into their own packages, so that we could use them in the types package when filling the DAH.

Reply:
I do think that once we move DAHedear calculation to `p2p/ipld` we can take the wrapper back to where it exists now in master.

Reply:
Having one more package for wrapper makes no sense for me in general

Reply:
Same for consts actually

Reply:
From experience I have in Golang: If I have an import cycle, it means I messed up proper logic segregation between packages architecturally, and simply extracting specific the piece of code causing the cycle is a simple hack, not a complete solution.

Reply:
I think you are tight about the import cycles and extraction into a separate package @Wondertan (in this case, not necessarily in general). 

IMO, we can simply revisit this when we move the computation of the DAHeader into PutBlock.